tests/rust/safe_abstraction/mutex_u32.rs

```diff
@@ -186,30 +196,120 @@ lem MutexU32_share_full(k: lifetime_t, t: thread_id_t, l: *MutexU32)
         }
     }
 }
+
+lem MutexU32_send(t: thread_id_t, t1: thread_id_t)
+    req MutexU32_own(t, ?inner, ?data);
+    ens MutexU32_own(t1, inner, data);
+{
+    open MutexU32_own(t, inner, data);
+    close MutexU32_own(t1, inner, data);
+}
+
+lem MutexU32_sync(t: thread_id_t, t1: thread_id_t)
```

Comment:
Whoops! Big proof! Not an ideal advertisement for our approach, I guess. An alternative might be to change the definition of MutexU32_share so that it doesn't use `t` at all and instead uses some dummy default thread id as the argument for u32_fbc? (You could use `default_value<thread_id_t>()`.) Then this proof would be trivial?

Reply:
It is possible to use a `pred_ctor u32_fbc(l: *u32)()` i.e. a `thread_id` independent version of `u32_full_borrow_content` and then make `MutexU32_frac_borrow_contetn` independent of `t`. but thinking about the generic version of `Mutex` there we should say that `T` is `Send`, do `[[T]].own(t, vs) -* [[T]].own(t1, vs)` and from that we can get the logical equivalence of full borrows of `T_full_borrow_content` with `t` and `t1` args. So the same proof pattern happens there and there would be the same big proof for that. Since `produce_lem_ptr_chunk` takes the chunks produced back, we need to duplicate equivalence proofs every time. If there is an alternative way to do that in VeriFast without duplication the proof would not be very long actually.

Reply:
I wonder if, as part of splitting the full borrow, we can rewrite `T_fbc(t, l)` to `T_fbc(default_value, l)`. But let's look at this later.